src/BlockArea.cpp

```diff
@@ -2572,14 +2572,7 @@ void cBlockArea::MergeByStrategy(const cBlockArea & a_Src, int a_RelX, int a_Rel
 			break;
 		}  // case msMask
 
-		#ifndef __clang__  // Clang complains about a default case in a switch with all cases covered
-		default:
-		{
-			LOGWARNING("Unknown block area merge strategy: %d", a_Strategy);
-			ASSERT(!"Unknown block area merge strategy");
-			return;
-		}
```

Comment:
I'm not that familar, what the `UNREACHABLE` macro does.
But I would like to specify the warning for each `COVERED_SWITCH`. Then users running a release build can also report a warning like: "Unknown block area merge strategy: 77", which helps more than no warning (changed behaviour) or the generic warning "Covered switch should never hit default".

Maybe a `COVERED_DEFAULT` macro would be more flexible. So that I can write something like this:

    COVERED_DEFAULT
    {
        LOGWARNING("We broke X")
        ASSERT(!"Should not happen")
        return KeepItWorkingValue;
    }

Reply:
If you look down at the bottom of these changes, UNREACHABLE is defined, and it does log an informative message.

Reply:
Oh, yeah. Then this is fine.

Reply:
>Maybe a `COVERED_DEFAULT` macro would be more flexible. So that I can write something like this:

A big part of this PR is trying to minimise the difference in behaviour between different platforms. As it stands many places are just a warning message on some platforms but undefined behaviour on others.  That seems very bad to me.

Reply:
I've just thought, that you could use the `#pragma` things on the macro to suppress Clang from giving the warnings, when using the macro... But sadly, it doesn't work.